Classes/ios/Scanners/MTBBarcodeScanner.m

```diff
@@ -175,6 +175,31 @@ + (BOOL)cameraIsPresent {
     return [AVCaptureDevice defaultDeviceWithMediaType:AVMediaTypeVideo] != nil;
 }
 
++ (BOOL)hasCamera:(MTBCamera)camera {
+    AVCaptureDevicePosition position = [self devicePositionForCamera:camera];
+
+    for (AVCaptureDevice *device in [AVCaptureDevice devicesWithMediaType:AVMediaTypeVideo]) {
+        if (device.position == position) {
+            return YES;
+        }
+    }
+
+    return NO;
+}
+
++ (MTBCamera)oppositeCameraOf:(MTBCamera)camera {
```

Comment:
Should not be a problem. The iOS 10 SDK allows you to use the dual camera system (as one device), but it will never be returned from the old APIs – you will need a special class  to access this type of camera:

 > **AVCaptureDeviceTypeBuiltInDualCamera**
 > A device that consists of two fixed focal length cameras, one wide and one telephoto. Note that devices of this type may only be discovered using an AVCaptureDeviceDiscoverySession.